tests/python/pants_test/backend/codegen/tasks/test_protobuf_integration.py

```diff
@@ -15,6 +15,12 @@
 
 class ProtobufIntegrationTest(PantsRunIntegrationTest):
 
+  def test_import_from_buildroot(self):
+    pants_run = self.run_pants(
+      ['gen.protoc', '--import-from-root',
+       'testprojects/src/protobuf/org/pantsbuild/testproject/import_from_buildroot/bar'])
+    self.assert_success(pants_run)
+
```

Comment:
Could you add a test that asserts that if there's a import for a proto file that is not owned by one of a target's dependencies that it fails? I didn't see one of those when I looked.

Maybe we can't ensure that right now, because of how source_roots work with targets.